src/System.Private.CoreLib/shared/System/Boolean.cs

```diff
@@ -175,6 +175,27 @@ public int CompareTo(Boolean value)
         // Static Methods
         // 
 
+        // Custom string compares for early application use by config switches, etc
+        // 
+        internal static bool IsTrueStringIgnoreCase(ReadOnlySpan<char> value)
+        {
+            return (value.Length == 4 &&
+                    (value[0] == 't' || value[0] == 'T') &&
```

Comment:
Probably not performance critical, but you half the number of branches by or ing with 0x20?
```csharp
(value[0] | 0x20) == 't'
(value[1] | 0x20) == 'r'
(value[2] | 0x20) == 'u'
(value[3] | 0x20) == 'e'
```

Reply:
Only about 1ns max there. Not much to the function I suppose.

Reply:
Furthermore, we might compare characters by pairs, e.g.,
```            ((*(int*)p ^ 0x520054) & ~0x200020) == 0
                && ((*((int*)p + 1) ^ 0x450055) & ~0x200020) == 0;```
where `p` points to `value[0]`.  While we could check all 4 characters simultaneously by reading a `long`, early bail out might be beneficial for 32-bit systems.